do_cmake.sh

```diff
@@ -6,24 +6,33 @@ if test -e build; then
 fi
 
 PYBUILD="2"
-source /etc/os-release
-case "$ID" in
-    fedora)
-        if [ "$VERSION_ID" -ge "29" ] ; then
-            PYBUILD="3"
-        fi
-        ;;
-    rhel|centos)
-        MAJOR_VER=$(echo "$VERSION_ID" | sed -e 's/\..*$//')
-        if [ "$MAJOR_VER" -ge "8" ] ; then
-            PYBUILD="3"
-        fi
-        ;;
-    opensuse*|suse|sles)
-        PYBUILD="3"
-        WITH_RADOSGW_AMQP_ENDPOINT="OFF"
-        ;;
-esac
+if [ -r /etc/os-release ]; then
+  source /etc/os-release
+  case "$ID" in
+      fedora)
+          if [ "$VERSION_ID" -ge "29" ] ; then
+              PYBUILD="3"
+          fi
+          ;;
+      rhel|centos)
+          MAJOR_VER=$(echo "$VERSION_ID" | sed -e 's/\..*$//')
+          if [ "$MAJOR_VER" -ge "8" ] ; then
+              PYBUILD="3"
+          fi
+          ;;
+      opensuse*|suse|sles)
+          PYBUILD="3"
+          WITH_RADOSGW_AMQP_ENDPOINT="OFF"
+          ;;
+  esac
+elif [ "$(uname)" == FreeBSD ]; then
```

Comment:
nit: please put a space before the semicolon, to match the style of the rest of the script?